### PR TITLE
ticket 6 - view a list of comments associated with a review

### PR DESCRIPTION
**Pull request by laumack**

### CODE REVIEW

File: src/components/Comments.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import "../styling/Comments.css";
+import { useEffect, useState } from "react";
+import { getComments } from "../utils.js";
+
+export default function Comments() {
+  const [commentsById, setComments] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getComments(2).then(({ comments }) => {
```
Comment (leekli) [inline src/components/Comments.jsx]: I think this is your culprit, you're always getting comments for id 2

**Review by leekli — CHANGES_REQUESTED** [review]
Hi Lauren,

I haven't been able to use your app properly - it all loads which is great, but it looks like it always loads the same comments for every review - looking at your code, I think the issue is in your `Comments.jsx` file line 10 where you're always passing the id 2.

Please could you fix this, test to make sure it works and then resubmit a PR.

To confirm (info for next tutor also) - I haven't properly tried the app, or reviewed any code so a full PR review will need completing once the resubmit comes in.

File: src/components/Comments.jsx (hunk 1 above)
Comment (laumack) [reply src/components/Comments.jsx]: Thank you, this has now been changed.

File: src/components/Comments.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { useParams } from "react-router-dom";
+import { getComments } from "../utils.js";
+
+export default function Comments() {
+  const { reviewId } = useParams();
+  const [commentsById, setCommentsById] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getComments(reviewId).then(({ comments }) => {
```
Comment (Jodiech) [inline src/components/Comments.jsx:12]: Nice work passing the reviewId in your useEffect 👍

File: src/components/Comments.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  if (commentsById.length === 0) {
+    return (
+      <>
+        <section className="comments-header">
+          <h2 id="title-comment">Comments</h2>
+          <h3 id="title-user">Username</h3>
+          <h3 id="title-votes">Votes</h3>
+        </section>
+        <p id="no-comments">No comments yet...</p>
```
Comment (Jodiech) [inline src/components/Comments.jsx:35]: I like that you have accounted for no comments and have some text for the user to see on the page 👍

File: src/components/ReviewById.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import "../styling/ReviewById.css";
+import { useEffect, useState } from "react";
+import { getReviewById } from "../utils.js";
+import { useParams } from "react-router-dom";
+import Comments from "./Comments.jsx";
+
+export default function ReviewById() {
+  const { reviewId } = useParams();
+  const [idReview, setIdReview] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
```
Comment (Jodiech) [inline src/components/ReviewById.jsx:10]: Great to see you have a loading state as a visual indicator to the user 🤸

File: src/components/ReviewById.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        ...loading...
+        <br />
+        please wait
+      </section>
+    );
+
+  return (
+    <main id="main">
+      <article className="review-container">
+        <h2 id="item-title">{idReview.title}</h2>
```
Comment (Jodiech) [inline src/components/ReviewById.jsx:31]: Just a suggestion here to alter your h2 for the **item-title** it seems to overlap with the rest of the text on the screen when viewed in mobile dimensions on the browser.

File: src/utils.js [hunk 6]
```diff
@@ -15,3 +15,9 @@ export function getReviewById(reviewId) {
     return res.data;
   });
 }
+
+export function getComments(reviewId) {
```
Comment (Jodiech) [inline src/utils.js:19]: Nice to see you are fetching the comments by review id 😄

File: src/App.jsx [hunk 7]
```diff
@@ -1,20 +1,22 @@
...
       <Nav />
-      <MainHeader />
+      <Header />
       <Routes>
         <Route path="/" element={<ReviewsList />}></Route>
-        <Route path="/reviews/:reviewId" element={<ReviewIdCard />}></Route>
+        <Route path="/reviews/:reviewId" element={<ReviewById />}></Route>
+        <Route path="/reviews/:reviewId" element={<Comments />}></Route>
       </Routes>
       <footer>Footer - under construction</footer>
```
Comment (Jodiech) [inline src/App.jsx:19]: I'd remove the footer if it is not yet being utilised. Not a problem, but when submitting PR's, it's preferable to include code that is fully implemented and ready for production use 👍

**Review by Jodiech — APPROVED** [review]
Hey 👋 good job on this ticket! I can see you have amended this to make the reviews render correctly on the page. Nicely done! I'm happy for you to merge this 🚀

File: src/App.jsx (hunk 7 above)
Comment (laumack) [reply src/App.jsx:19]: That's good to know, thanks Jodie

**Pull request merged**